### ISSUE TITLE
Non zero diameter for degenerated interval

### ISSUE THREAD

**Issue by d-monnet** [JuliaIntervals/IntervalArithmetic.jl#545]
Hi there,
Is there a reason why the diameter of a degenerated interval is the smallest representable float and not 0 ?
```julia
x = Interval(1) # degenerated interval
x.lo === x.hi # bounds are the same
diam(x) === 0.0 # false
diam(x) === nextfloat(0.0) # true
```
Would it be possible to add something like 
```julia
if x.lo === x.hi
  return 0
end
``` 
somewhe in `diam` implementation ?

**Comment by lbenet** [comment]
Hi! Thanks for opening this. I can't reproduce the issue you report, so maybe we have different versions. I'm using Julia v1.8.2, and IntervalArithmetics v 0.20.7. This is the output I get:
```julia
julia> using IntervalArithmetic

julia> x = Interval(1)
[1, 1]

julia> x.lo === x.hi
true

julia> diam(x) === 0.0
true
```

**Comment by lucaferranti** [comment]
there are situations in which that behavior is expected and (depending on the philosophy) even desired. For example

```julia
julia> a = 0.1..0.1
[0.0999999, 0.100001]

julia> diam(a)
1.3877787807814457e-17
```

the reason for this is that `0.1` is not exactly representable as floating point number and the `..` constructor (and the `@interval`) macro have some heuristic to detect when the typed number is exactly representable of not. If it looks like it doesn't, they will return an interval with non-zero diameter, which contains the true `0.1` number

```julia
julia> big"0.1" ∈ a
true
```

If the algorithm behind the scenes decides that the typed number is probably exactly representable, then it will return an interval with zero-width

```julia
julia> diam(1..1)
0.0

julia> diam(0.5..0.5)
0.0
```

Your example is strange, because 

1) `1..1` should not widen the interval, as my example above shows
2) `Interval` (with capital I) is an unsafe constructor, meant to be used internally for efficiency, which does not perform any checks or any heuristics

could you type the following in the REPL

```julia
using Pkg; Pkg.status("IntervalArithmetic")
```
and paste the output?

**Comment by d-monnet** [comment]
I used v0.20.6. The problem is solved with v0.20.7 ! Sorry for not checking the version first.
